### ISSUE TITLE
How to evaluate the ciphertext size after encrypting with CKKS scheme?

### ISSUE THREAD

**Issue by napoay** [ibarrond/Pyfhel#158]
**How to evaluate the ciphertext size after encrypting with CKKS scheme?**

Since N/2 values will be encoded in a single ciphertext, how can I compare the ciphertext size of V1 (length a) and V2 (length b)? Where a < b < N/2.

**Test Code:**
```

import numpy as np
from Pyfhel import Pyfhel

HE = Pyfhel()           # Creating empty Pyfhel object
ckks_params = {
    'scheme': 'CKKS',   # can also be 'ckks'
    'n': 2**15,         # Polynomial modulus degree. For CKKS, n/2 values can be
                        #  encoded in a single ciphertext. 
                        #  Typ. 2^D for D in [10, 16]
    'scale': 2**30 ,    # All the encodings will use it for float->fixed point
                        #  conversion: x_fix = round(x_float * scale)
                        #  You can use this as default scale or use a different
                        #  scale on each operation (set in HE.encryptFrac)
    'qi_sizes': [60, 30, 30, 30, 60] # Number of bits of each prime in the chain. 
                        # Intermediate values should be  close to log2(scale)
                        # for each operation, to have small rounding errors.
}

ckks_params = {'scheme': 'CKKS', 'n': 2**12,  'scale': 2**30,  'qi_sizes': [30, 30, 30]}

HE.contextGen(**ckks_params)  # Generate context for bfv scheme
HE.keyGen()             # Key Generation: generates a pair of public/secret keys
HE.rotateKeyGen()

v1 = np.random.random(1000)
v2 = np.random.random(2000)


ptxt_x = HE.encodeFrac(v1)   
ptxt_y = HE.encodeFrac(v2)   

ctxt_v1 = HE.encryptPtxt(ptxt_x)
ctxt_v2 = HE.encryptPtxt(ptxt_y) 

tmp_dir_name = "/home/paan/Desktop/tmp_dir"

ctxt_v1.save(tmp_dir_name + "/ctxt_v1.ptxt")

ctxt_v2.save(tmp_dir_name + "/ctxt_v2.ptxt")
```
The sizes of `ctxt_v1.ptxt ` and `ctxt_v2.ptxt` are the same. Can the size of these two files be used as the ciphertext size of the corresponding plaintext array?

**Comment by ibarrond** [comment]
I'm afraid not! Leaking the "size" (number of non-zero encrypted values) of a ciphertext would completely break the privacy-preserving properties of FHE schemes, since you could always multiply a ciphertext with many carefully selected plaintexts to extract the element-wise information based on the changes on the "size".

That being said, if we implement the homomorphic comparison (#115), you could turn all non-zero values inside a ciphertext into ones and perform a cumulative sum to get the (encrypted) number of non-zero elements in the original vector. Sadly the comparison is not there quite yet!

**Comment by ibarrond** [comment]
Yes! That can be seen as the communication cost.

Related to #134. All the `save` operations return the size (in bytes) of the saved object. Additionally, as per 8ae1a2ccba61884881d02c93ad50b60559e03dfd (currently in the `master` branch, to be included in `Pyfhel==3.3.2` in the near future), there are multiple `sizeof` functions that you can call to get that same size of every object without actually storing it.

**Comment by napoay** [comment]
Thanks a bunch！ @ibarrond
